includes/install.php

```diff
@@ -515,14 +515,14 @@ function give_install_tables_on_plugin_update($old_version)
 function __give_get_tables()
 {
     $tables = [
-        'donors_db' => new Give_DB_Donors(),
-        'donor_meta_db' => new Give_DB_Donor_Meta(),
-        'comment_db' => new Give_DB_Comments(),
-        'comment_db_meta' => new Give_DB_Comment_Meta(),
-        'give_session' => new Give_DB_Sessions(),
-        'formmeta_db' => new Give_DB_Form_Meta(),
-        'sequential_db' => new Give_DB_Sequential_Ordering(),
-        'donation_meta' => new Give_DB_Payment_Meta(),
+        'donors_db' => give()->get(Give_DB_Donors::class),
+        'donor_meta_db' => give()->get(Give_DB_Donor_Meta::class),
+        'comment_db' => give()->get(Give_DB_Comments::class),
+        'comment_db_meta' => give()->get(Give_DB_Comment_Meta::class),
+        'give_session' => give()->get(Give_DB_Sessions::class),
+        'formmeta_db' => give()->get(Give_DB_Form_Meta::class),
+        'sequential_db' => give()->get(Give_DB_Sequential_Ordering::class),
+        'donation_meta' => give()->get(Give_DB_Payment_Meta::class),
```

Comment:
What's happening here?

Reply:
All of these classes are extensions of `Give_DB_Meta`. 

However, this class, during its instantiation, will override all post modification methods via hooks such as:

```php
if (in_array('delete_post_metadata', $this->supports)) {
    add_filter('delete_post_metadata', [$this, '__delete_meta'], 0, 5);
}
```

This will trigger a bugged behavior if one of these classes is instantiated more than once.

In the specific case of deletion, the return value indicates the number of rows deleted from the database.

But if the hook is called by several instances of the class, this is what will happen:

1. The first hook will do the deletion work and return the number of rows deleted
2. The second hook will look for lines to delete, find none (they were all deleted by the first hook) and therefore return the value `0`
3. The hook ends by systematically returning `0`, the return value is wrong.

For this reason, I made the minimum possible modifications to approach singleton pattern behavior.

This is the code you present here.

Reply:
🤔 interesting!

I think the singleton pattern here does make sense 👍 

We'll just have to run it through QA internally to make sure everything works as expected.